### PR TITLE
istioctl dashboard bind all addresses

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Please provide a description for what this PR is for.

This is to fix this case: the host that running `istioctl dashboard xxx ` does not install a browser

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[X] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
@esnible
I think istioctl dashboard is mainly a debug tool. We donot need to run it all the way. If you literally think it is worth adding a new flag, i can do that.

**Comment by myidpt** [comment]
Yes, please at least add a new flag for this feature. Opening 0.0.0.0 instead of localhost by default is a security vulnerability.

**Comment by hzxuzhonghu** [comment]
Thanks, will do

File: istioctl/cmd/dashboard.go [hunk 1]
```diff
@@ -355,6 +357,11 @@ func dashboard() *cobra.Command {
 		},
 	}
 
+	dashboardCmd.PersistentFlags().StringVar(&bindAddress, "address", "localhost",
+		"Addresse to listen on. Only accepts IP address or localhost as a value. "+
```
Comment (esnible) [inline istioctl/cmd/dashboard.go]: Spelling of "addresse"

**Comment by hzxuzhonghu** [comment]
I do not change any behavior, the `localhost` is the default value used by kubectl and istioctl.

File: istioctl/pkg/kubernetes/client.go [hunk 2]
```diff
@@ -348,7 +348,10 @@ func (client *Client) BuildPortForwarder(podName string, ns string, localPort in
 
 	stop := make(chan struct{})
 	ready := make(chan struct{})
-	fw, err := portforward.New(dialer, []string{fmt.Sprintf("%d:%d", localPort, podPort)}, stop, ready, ioutil.Discard, os.Stderr)
+	if localAddr == "" {
+		localAddr = "localhost"
+	}
+	fw, err := portforward.NewOnAddresses(dialer, []string{localAddr}, []string{fmt.Sprintf("%d:%d", localPort, podPort)}, stop, ready, ioutil.Discard, os.Stderr)
```
Comment (hzxuzhonghu) [inline istioctl/pkg/kubernetes/client.go]: @esnible 

```
// New creates a new PortForwarder with localhost listen addresses.
func New(dialer httpstream.Dialer, ports []string, stopChan <-chan struct{}, readyChan chan struct{}, out, errOut io.Writer) (*PortForwarder, error) {
	return NewOnAddresses(dialer, []string{"localhost"}, ports, stopChan, readyChan, out, errOut)
}

```

**Pull request merged**